In [108]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from sqlalchemy import create_engine
import psycopg2 as pg

Ćwiczenie 3

In [109]:
# db_string = "postgres://postgres:admin@localhost:5432/test"
# db = create_engine(db_string)
# connection_sqlalchemy = db.connect()
from multiprocessing import connection


connection = pg.connect(host='pgsql-196447.vipserv.org', port=5432, dbname='wbauer_adb', user='wbauer_adb', password='adb2020')



### 1. Liczba kategorii filmów. ###

In [110]:
#1
sl = pd.read_sql('SELECT \
    COUNT(name) \
        FROM category', \
            con=connection)
print(sl)



   

   count
0     16


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/2968672298.py:2: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  sl = pd.read_sql('SELECT \


W wypożyczali jest 16 kategorii filmów. 

### 2. LIsta kategorii w kolejności alfabetycznej.  ###

In [111]:
#2
s2 = pd.read_sql('SELECT name \
    FROM category \
        ORDER BY name \
            ASC', con=connection)
print(s2)

           name
0        Action
1     Animation
2      Children
3      Classics
4        Comedy
5   Documentary
6         Drama
7        Family
8       Foreign
9         Games
10       Horror
11        Music
12          New
13       Sci-Fi
14       Sports
15       Travel


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1239474922.py:2: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s2 = pd.read_sql('SELECT name \


### 3. Najstarszy i najmłodszy film w wypożyczalni.  ###

In [112]:
#3
s3 = pd.read_sql('SELECT title, \
    release_year FROM film \
        ORDER BY release_year \
            ASC \
                LIMIT 1', \
                    con=connection)
print(s3)
s4 = pd.read_sql('SELECT title, release_year FROM film ORDER BY release_year DESC LIMIT 1', con=connection)
print(s4)

C:\Users\majga\AppData\Local\Temp/ipykernel_16172/2317401092.py:2: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s3 = pd.read_sql('SELECT title, \


             title  release_year
0  Chamber Italian          2006
             title  release_year
0  Chamber Italian          2006


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/2317401092.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s4 = pd.read_sql('SELECT title, release_year FROM film ORDER BY release_year DESC LIMIT 1', con=connection)


Najmłodszy i najstarszy film pochodzą z 2006 roku. 

### 4. Liczba wypożyczeń w danym okresie czasu.  ###

In [113]:
s4 = pd.read_sql(
    "SELECT COUNT(rental_date) \
        FROM rental \
            WHERE rental_date >= '2005-07-01 00:00:00' and rental_date <= '2005-08-01 23:59:59'", \
                con=connection)
print(s4)

   count
0   7380


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/2927732750.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s4 = pd.read_sql(


W tym okresie zostało wypożyczonych 7380 filmów. 

### 5. Liczba wypożyczeń w danym okresie czasu.  ###

In [114]:
s5 = pd.read_sql(
    "SELECT COUNT(rental_date) \
        FROM rental \
            WHERE rental_date >= '2010-01-01 00:00:00' and rental_date <= '2011-02-01 23:59:59'", \
                con=connection)
print(s5)

C:\Users\majga\AppData\Local\Temp/ipykernel_16172/2162587030.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s5 = pd.read_sql(


   count
0      0


W tym okresie zostało wypożyczonych 0 filmów.  

### 6. Największa płatność wypożyczenia  ###

In [115]:
#6
s6 = pd.read_sql('SELECT * \
    FROM payment \
        ORDER BY amount \
            DESC \
                LIMIT 1', \
                    con=connection)
print(s6)

   payment_id  customer_id  staff_id  rental_id  amount  \
0       20403          362         1      14759   11.99   

                payment_date  
0 2007-03-21 21:57:24.996577  


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1571227255.py:2: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s6 = pd.read_sql('SELECT * \


### 7. Klienci z Polka, Nigeria lub Bangladesz. ###

In [116]:
#7



s7 = pd.read_sql("SELECT country_id \
    FROM country\
        WHERE country = 'Poland' OR country = 'Nigeria' OR country = 'Bangladesh'\
    ", con=connection) #Znalezienie id krajów
s7_7 = pd.read_sql("SELECT city_id\
    FROM city where country_id ='12' OR country_id ='69' OR country_id ='76'"\
        ,con=connection) #Znalezienie id krajów na popdstawie id miast
s7_7_tp = tuple(s7_7["city_id"].values.tolist()) 
address_id = pd.read_sql("SELECT address_id\
    FROM address\
        WHERE city_id IN {}".format(s7_7_tp),\
            con=connection) #Znalezienie id adresów na podstawie id miast

address_id_tp = tuple(address_id["address_id"].values.tolist())

customer = pd.read_sql("SELECT *\
    FROM customer\
        WHERE address_id IN {}".format(address_id_tp), \
            con=connection) #znalezienie klientów na podstawie id adresów
customer


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1516522755.py:5: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s7 = pd.read_sql("SELECT country_id \
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1516522755.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  s7_7 = pd.read_sql("SELECT city_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1516522755.py:13: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  address_id = pd.read_sql("SELECT address_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1516522755.py:20: Us

customer_id  store_id first_name  last_name  \
0            18         2      Carol     Garcia   
1            21         1   Michelle      Clark   
2            80         1    Marilyn       Ross   
3           103         1     Gladys   Hamilton   
4           128         1   Marjorie     Tucker   
5           156         1     Bertha   Ferguson   
6           198         2      Elsie     Kelley   
7           232         2  Constance       Reid   
8           250         2         Jo     Fowler   
9           270         1       Leah     Curtis   
10          277         2       Olga    Jimenez   
11          280         2     Tracey    Barrett   
12          284         1      Sonia    Gregory   
13          286         1      Velma      Lucas   
14          318         1      Brian      Wyman   
15          329         2      Frank   Waggoner   
16          332         1    Stephen     Qualls   
17          380         1    Russell    Brinson   
18          409         2     Rodney    Moeller   
19          501         1      Ruben      Geary   
20          558         1     Jimmie  Eggleston   
21          562         1    Wallace      Slone   
22          571         2    Johnnie   Chisholm   
23          572         1     Sidney   Burleson   

                                  email  address_id  activebool create_date  \
0       carol.garcia@sakilacustomer.org          22        True  2006-02-14   
1     michelle.clark@sakilacustomer.org          25        True  2006-02-14   
2       marilyn.ross@sakilacustomer.org          84        True  2006-02-14   
3    gladys.hamilton@sakilacustomer.org         107        True  2006-02-14   
4    marjorie.tucker@sakilacustomer.org         132        True  2006-02-14   
5    bertha.ferguson@sakilacustomer.org         160        True  2006-02-14   
6       elsie.kelley@sakilacustomer.org         202        True  2006-02-14   
7     constance.reid@sakilacustomer.org         236        True  2006-02-14   
8          jo.fowler@sakilacustomer.org         254        True  2006-02-14   
9        leah.curtis@sakilacustomer.org         275        True  2006-02-14   
10      olga.jimenez@sakilacustomer.org         282        True  2006-02-14   
11    tracey.barrett@sakilacustomer.org         285        True  2006-02-14   
12     sonia.gregory@sakilacustomer.org         289        True  2006-02-14   
13       velma.lucas@sakilacustomer.org         291        True  2006-02-14   
14       brian.wyman@sakilacustomer.org         323        True  2006-02-14   
15    frank.waggoner@sakilacustomer.org         334        True  2006-02-14   
16    stephen.qualls@sakilacustomer.org         337        True  2006-02-14   
17   russell.brinson@sakilacustomer.org         385        True  2006-02-14   
18    rodney.moeller@sakilacustomer.org         414        True  2006-02-14   
19       ruben.geary@sakilacustomer.org         506        True  2006-02-14   
20  jimmie.eggleston@sakilacustomer.org         564        True  2006-02-14   
21     wallace.slone@sakilacustomer.org         568        True  2006-02-14   
22  johnnie.chisholm@sakilacustomer.org         577        True  2006-02-14   
23   sidney.burleson@sakilacustomer.org         578        True  2006-02-14   

               last_update  active  
0  2013-05-26 14:49:45.738       1  
1  2013-05-26 14:49:45.738       1  
2  2013-05-26 14:49:45.738       1  
3  2013-05-26 14:49:45.738       1  
4  2013-05-26 14:49:45.738       1  
5  2013-05-26 14:49:45.738       1  
6  2013-05-26 14:49:45.738       1  
7  2013-05-26 14:49:45.738       1  
8  2013-05-26 14:49:45.738       1  
9  2013-05-26 14:49:45.738       1  
10 2013-05-26 14:49:45.738       1  
11 2013-05-26 14:49:45.738       1  
12 2013-05-26 14:49:45.738       1  
13 2013-05-26 14:49:45.738       1  
14 2013-05-26 14:49:45.738       1  
15 2013-05-26 14:49:45.738       1  
16 2013-05-26 14:49:45.738       1  
17 2013-05-26 14:49:45.738       1  
18 2013-05-26 14:49:45.738       1  
19 2013-05-2

### 8. Miejsce zamieszkania członków persolenu ### 

In [117]:
add_id = pd.read_sql("SELECT address_id \
    FROM staff"\
        , con=connection) #Znalezienie id krajów
add_id_tp = tuple(add_id["address_id"].values.tolist()) 
address = pd.read_sql("SELECT address, address2\
    FROM address WHERE address_id IN {}".format(add_id_tp)\
        ,con=connection) 

address





C:\Users\majga\AppData\Local\Temp/ipykernel_16172/316211809.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  add_id = pd.read_sql("SELECT address_id \
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/316211809.py:5: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  address = pd.read_sql("SELECT address, address2\


address address2
0     23 Workhaven Lane     None
1  1411 Lillydale Drive     None

### 9. Liczba pracowników w Argentynie lub Hiszpani ###

In [118]:
city_id = pd.read_sql("SELECT city_id\
    FROM address WHERE address_id IN {}".format(add_id_tp)\
        ,con=connection)
city_id_tp = tuple(city_id["city_id"].values.tolist()) 

country_id = pd.read_sql("SELECT country_id\
    FROM city WHERE city_id IN {}".format(city_id_tp)\
        ,con=connection)
country_id_tp = tuple(country_id["country_id"].values.tolist()) 

country = pd.read_sql("SELECT COUNT(*)\
    FROM country WHERE (country = 'Argentina' OR country = 'Spain' ) AND country_id IN {}".format(country_id_tp)\
        ,con=connection)
country


C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3815125141.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  city_id = pd.read_sql("SELECT city_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3815125141.py:6: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  country_id = pd.read_sql("SELECT country_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3815125141.py:11: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  country = pd.read_sql("SELECT COUNT(*)\


count
0      0

### 10. Kategorie wypożyczanych filmów ###

In [119]:
rental_id = pd.read_sql("SELECT rental_id\
    FROM payment"\
        ,con=connection)
rental_id_tp = tuple(rental_id["rental_id"].values.tolist()) 

inventory_id = pd.read_sql("SELECT inventory_id\
    FROM rental WHERE rental_id IN {}".format(rental_id_tp)\
        ,con=connection)
inventory_id_tp = tuple(inventory_id["inventory_id"].values.tolist()) 

film_id = pd.read_sql("SELECT film_id\
    FROM inventory WHERE inventory_id IN {}".format(inventory_id_tp)\
        ,con=connection)
film_id_tp = tuple(film_id["film_id"].values.tolist()) 
category_id = pd.read_sql("SELECT category_id\
    FROM film_category WHERE film_id IN {}".format(film_id_tp)\
        ,con=connection)
category_id_tp = tuple(category_id["category_id"].values.tolist()) 


name = pd.read_sql("SELECT name\
    FROM category WHERE category_id IN {}".format(category_id_tp)\
        ,con=connection)

name

C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1689987539.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  rental_id = pd.read_sql("SELECT rental_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1689987539.py:6: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  inventory_id = pd.read_sql("SELECT inventory_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/1689987539.py:11: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  film_id = pd.read_sql("SELECT film_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/16899875

name
0        Action
1     Animation
2      Children
3      Classics
4        Comedy
5   Documentary
6         Drama
7        Family
8       Foreign
9         Games
10       Horror
11        Music
12          New
13       Sci-Fi
14       Sports
15       Travel

### 11.Kategorie filmów wypożyczonych w USA ###

In [120]:
cou_id = pd.read_sql("SELECT country_id\
    FROM country WHERE country = 'United States'"\
        ,con=connection)
cou_id_int = int(cou_id["country_id"].values)

city_id = pd.read_sql("SELECT city_id FROM city WHERE country_id = {}".format(cou_id_int),con=connection)
city_id_tp = tuple(city_id["city_id"].values.tolist()) 

adres_id = pd.read_sql("SELECT address_id\
    FROM address WHERE city_id IN {}".format(city_id_tp)\
        ,con=connection)
adres_id_tp = tuple(adres_id["address_id"].values.tolist())

customer_id = pd.read_sql("SELECT customer_id\
    FROM customer WHERE address_id IN {}".format(adres_id_tp)\
        ,con=connection)
customer_id_tp = tuple(customer_id["customer_id"].values.tolist()) 


inventory_id = pd.read_sql("SELECT inventory_id\
    FROM rental WHERE customer_id IN {}".format(customer_id_tp)\
        ,con=connection)
inventory_id_tp = tuple(inventory_id["inventory_id"].values.tolist()) 


film_id = pd.read_sql("SELECT film_id\
    FROM inventory WHERE inventory_id IN {}".format(inventory_id_tp)\
        ,con=connection)
film_id_tp = tuple(film_id["film_id"].values.tolist()) 


cat_id = pd.read_sql("SELECT category_id\
    FROM film_category WHERE film_id IN {}".format(film_id_tp)\
        ,con=connection)
cat_id_tp = tuple(cat_id["category_id"].values.tolist()) 


name = pd.read_sql("SELECT name\
    FROM category WHERE category_id IN {}".format(cat_id_tp)\
        ,con=connection)

name

C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3861820629.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  cou_id = pd.read_sql("SELECT country_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3861820629.py:6: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  city_id = pd.read_sql("SELECT city_id FROM city WHERE country_id = {}".format(cou_id_int),con=connection)
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3861820629.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  adres_id = pd.read_sql("SELECT address_id\


name
0        Action
1     Animation
2      Children
3      Classics
4        Comedy
5   Documentary
6         Drama
7        Family
8       Foreign
9         Games
10       Horror
11        Music
12          New
13       Sci-Fi
14       Sports
15       Travel

### 12. Znajdź wszystkie tytuły filmów, w których grał: Olympia Pfeiffer lub Julia Zellweger lub Ellen Presley ###

In [121]:
actor_id = pd.read_sql("SELECT actor_id\
    FROM actor WHERE (first_name = 'Olympia' AND last_name = 'Pfeiffer') OR (first_name = 'Julia' AND last_name = 'Zellweger') OR (first_name = 'Ellen' AND last_name = 'Presley')"\
           ,con=connection)
actor_id_tp = tuple(actor_id["actor_id"].values)

film_id = pd.read_sql("SELECT film_id FROM film_actor WHERE actor_id IN {}".format(actor_id_tp),con=connection)
film_id_tp = tuple(film_id["film_id"].values.tolist()) 

title = pd.read_sql("SELECT title\
    FROM film WHERE film_id IN {}".format(film_id_tp)\
        ,con=connection)
title

C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3096292282.py:1: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  actor_id = pd.read_sql("SELECT actor_id\
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3096292282.py:6: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  film_id = pd.read_sql("SELECT film_id FROM film_actor WHERE actor_id IN {}".format(actor_id_tp),con=connection)
C:\Users\majga\AppData\Local\Temp/ipykernel_16172/3096292282.py:9: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  title = pd.read_sql("SELECT title\


title
0            Badman Dawn
1        Bilko Anonymous
2   Breakfast Goldfinger
3      Caribbean Liberty
4       Casper Dragonfly
..                   ...
64             Turn Star
65            Wait Cider
66          Women Dorado
67             Won Dares
68         Wyoming Storm

[69 rows x 1 columns]